In [1]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


In [2]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import warnings

from src.features import build_features

warnings.filterwarnings("ignore")
sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_DEL_TORAX = 112103

## 1. Análisis de Procedimientos a Nivel País

En este análisis se quieren responder las siguiente preguntas:

1. ¿Cómo es el ranking de los hospitales para cada uno de los procedimientos realizads?
2. ¿Quién es el Hospital que realiza más de X procedimientos?
3. ¿El procedimiento X es realizado en todo el país? ¿En qué partes en específico se realiza?

Para responder esta pregunta se utilizará la base de datos de Egresos de Hospitales adscritos
a GRD entre 2019 y 2021.


In [3]:
vars_procedimientos = {f"PROCEDIMIENTO{i}": str for i in range(1, 31)}

In [4]:
df_procesada = pd.read_csv("../data/processed/df_procesada.csv", sep=";", dtype=vars_procedimientos)

In [5]:
display(df_procesada.head(10))

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO  ETNIA   PROVINCIA  \
0        118100       1314867.0   MUJER       1988-06-19  OTRO   CONCEPCION   
1        118100       1354418.0   MUJER       1988-03-27  OTRO   CONCEPCION   
2        118100        239861.0   MUJER       2017-06-17  OTRO   CONCEPCION   
3        118100        330326.0  HOMBRE       2017-12-04  OTRO   CONCEPCION   
4        118100       1369293.0   MUJER       1995-12-06  OTRO   CONCEPCION   
5        118100        469385.0   MUJER       1942-06-29  OTRO   CONCEPCION   
6        118100       1146635.0   MUJER       1951-07-14  OTRO   CONCEPCION   
7        118100         93557.0  HOMBRE       1938-03-04  OTRO   CONCEPCION   
8        118100        674943.0   MUJER       1962-08-21  OTRO   CONCEPCION   
9        118100       1079955.0  HOMBRE       1944-12-07  OTRO   CONCEPCION   

                COMUNA NACIONALIDAD                       PREVISION  \
0  SAN PEDRO DE LA PAZ        CHILE  FONASA INSTITUCIONAL - (MAI) A   
1  SAN PEDRO DE LA PAZ        CHILE  FONASA INSTITUCIONAL - (MAI) D   
2           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   
3           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) A   
4              FLORIDA        CHILE  FONASA INSTITUCIONAL - (MAI) C   
5          CHIGUAYANTE        CHILE  FONASA INSTITUCIONAL - (MAI) B   
6           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   
7                 LOTA        CHILE  FONASA INSTITUCIONAL - (MAI) D   
8          SANTA JUANA        CHILE  FONASA INSTITUCIONAL - (MAI) B   
9           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   

  SERVICIO_SALUD  ... USOSPABELLON IR_29301_COD_GRD IR_29301_PESO  \
0     CONCEPCIÓN  ...          1.0         146102.0        0.5744   
1     CONCEPCIÓN  ...          NaN         134161.0        0.2951   
2     CONCEPCIÓN  ...          NaN          64182.0        0.6736   
3     CONCEPCIÓN  ...          NaN          64172.0        0.5475   
4     CONCEPCIÓN  ...          NaN         134141.0        0.3107   
5     CONCEPCIÓN  ...          NaN         114103.0        1.1799   
6     CONCEPCIÓN  ...          NaN          44163.0        1.0328   
7     CONCEPCIÓN  ...          1.0          11101.0        1.8731   
8     CONCEPCIÓN  ...          1.0          61402.0        1.3078   
9     CONCEPCIÓN  ...          NaN          14133.0        1.6687   

  IR_29301_SEVERIDAD IR_29301_MORTALIDAD  \
0                2.0                 1.0   
1                1.0                 1.0   
2                2.0                 2.0   
3                2.0                 2.0   
4                1.0                 1.0   
5                3.0                 3.0   
6                3.0                 3.0   
7                1.0                 1.0   
8                2.0                 2.0   
9                3.0                 3.0   

                          HOSPPROCEDENCIA ESTANCIA ANIO_EGRESO MES_EGRESO  \
0                                     NaN      5.0      2019.0        3.0   
1                                     NaN      4.0      2019.0        3.0   
2                                     NaN      6.0      2019.0        3.0   
3                                     NaN      2.0      2019.0        3.0   
4                                     NaN      6.0      2019.0        3.0   
5                                     NaN     47.0      2019.0        3.0   
6                                     NaN     43.0      2019.0        3.0   
7                                     NaN     41.0      2019.0        3.0   
8  HOSPITAL CLORINDA AVELLO (SANTA JUANA)     33.0      2019.0        3.0   
9                                     NaN     31.0      2019.0        3.0   

    FECHA  
0  2019-3  
1  2019-3  
2  2019-3  
3  2019-3  
4  2019-3  
5  2019-3  
6  2019-3  
7  2019-3  
8  2019-3  
9  2019-3  

[10 rows x 133 columns]

Los procedimientos en esta base de datos se encuentran en las columnas "PROCEDIMIENTOSX", donde X
va desde 1 hasta 30. Para analizar esta base de forma más expedita, se transformará de formato
ancho a formato largo, utilizando como identificador a las columnas ["COD_HOSPITAL", "ANIO_EGRESO",
"CIP_ENCRIPTADO", "DIAGNOSTICO1"]. Cabe destacar que la columna "CIP_ENCRIPTADO" hace referencia al
RUT del paciente.


In [6]:
procedimientos_formato_long = pd.melt(
    df_procesada,
    id_vars=[
        "COD_HOSPITAL",
        "ANIO_EGRESO",
        "TIPO_ACTIVIDAD",
        "CIP_ENCRIPTADO",
        "DIAGNOSTICO1",
    ],
    value_vars=vars_procedimientos.keys(),
    value_name="procedimiento",
)

In [7]:
var_categorica_a_contar = "procedimiento"
desglose_a_obtener = ["ANIO_EGRESO", "COD_HOSPITAL"]
grupo_ranking = ["ANIO_EGRESO", "procedimiento"]

conteo_y_ranking_procedimientos = (
    build_features.calcular_conteo_y_ranking_sobre_variable_categorica(
        procedimientos_formato_long,
        var_categorica_a_contar,
        desglose_a_obtener,
        grupo_ranking,
    )
)

In [8]:
display(conteo_y_ranking_procedimientos.head(20))

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo_procedimiento  \
180193       2021.0        113180         99.99                     4   
160738       2021.0        108100         99.99                     2   
148451       2021.0        103100         99.99                     1   
152391       2021.0        105100         99.99                     1   
166209       2021.0        110100         99.99                     1   
167455       2021.0        110120         99.99                     1   
179376       2021.0        113130         99.99                     1   
182902       2021.0        114101         99.99                     1   
187970       2021.0        115107         99.99                     1   
188758       2021.0        115110         99.99                     1   
197375       2021.0        118100         99.99                     1   
202548       2021.0        121109         99.99                     1   
206349       2021.0        122100         99.99                     1   
211747       2021.0        126100         99.99                     1   
213212       2021.0        129106         99.98                    47   
206598       2021.0        123100         99.98                    25   
162279       2021.0        109100         99.98                    16   
180013       2021.0        113180         99.98                    15   
165111       2021.0        110100         99.98                    14   
196061       2021.0        118100         99.98                    14   

        ranking_conteo_procedimiento  diferencias_procedimiento  \
180193                             1                          2   
160738                             2                          1   
148451                             3                          0   
152391                             4                          0   
166209                             5                          0   
167455                             6                          0   
179376                             7                          0   
182902                             8                          0   
187970                             9                          0   
188758                            10                          0   
197375                            11                          0   
202548                            12                          0   
206349                            13                          0   
211747                            14                        -46   
213212                             1                         22   
206598                             2                          9   
162279                             3                          1   
180013                             4                          1   
165111                             5                          0   
196061                             6                          0   

        totales_procedimiento  proporcion_procedimiento  
180193                     18                      0.22  
160738                     18                      0.11  
148451                     18                      0.06  
152391                     18                      0.06  
166209                     18                      0.06  
167455                     18                      0.06  
179376                     18                      0.06  
182902                     18                      0.06  
187970                     18                      0.06  
188758                     18                      0.06  
197375                     18                      0.06  
202548                     18                      0.06  
206349                     18                      0.06  
211747                     18                      0.06  
213212                    235                      0.20  
206598                    235                      0.11  
162279                    235                      0.07  
180013                

El resultado muestra el ranking para todos los procedimientos asignados en el país en cada uno de
los años analizados (2019 a 2021).

A modo de ejemplo, se ilustra el ranking en el año 2021 para el procedimiento 99.99. En este caso,
se observa que los hospitales con código 108100 y 113180 están en los primeros lugares, realizando
2 procedimientos cada uno.

Ahora, si se filtra por el hospital del tórax, y sólo los procedimientos donde el hospital sea
ranking 1, entonces se obtendrán procedimientos altamente relevantes para el país.


In [9]:
cie_9 = pd.read_excel("../data/external/CIE-9.xlsx", dtype={"Código": str})
esquema_hospitales = pd.read_excel(
    "../data/external/Esquema_Registro-2023.xls", sheet_name=4, header=7
).dropna(how="all", axis=1)

In [10]:
ranking_con_nombre_procedimientos = pd.merge(
    conteo_y_ranking_procedimientos, cie_9, how="left", left_on="procedimiento", right_on="Código"
)
ranking_con_nombre_procedimientos_y_establecimientos = ranking_con_nombre_procedimientos.merge(
    esquema_hospitales, how="left", left_on="COD_HOSPITAL", right_on="Código nuevo Establecimiento"
)

In [11]:
display(ranking_con_nombre_procedimientos_y_establecimientos)

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo_procedimiento  \
0            2021.0        113180         99.99                     4   
1            2021.0        108100         99.99                     2   
2            2021.0        103100         99.99                     1   
3            2021.0        105100         99.99                     1   
4            2021.0        110100         99.99                     1   
...             ...           ...           ...                   ...   
214751       2019.0        105101         00.01                    65   
214752       2019.0        106100         00.01                    13   
214753       2019.0        113100         00.01                     1   
214754       2019.0        120101         00.01                     1   
214755       2019.0        114105             0                     2   

        ranking_conteo_procedimiento  diferencias_procedimiento  \
0                                  1                          2   
1                                  2                          1   
2                                  3                          0   
3                                  4                          0   
4                                  5                          0   
...                              ...                        ...   
214751                             1                         52   
214752                             2                         12   
214753                             3                          0   
214754                             4                         -1   
214755                             1                       <NA>   

        totales_procedimiento  proporcion_procedimiento Código  \
0                          18                      0.22  99.99   
1                          18                      0.11  99.99   
2                          18                      0.06  99.99   
3                          18                      0.06  99.99   
4                          18                      0.06  99.99   
...                       ...                       ...    ...   
214751                     80                      0.81  00.01   
214752                     80                      0.16  00.01   
214753                     80                      0.01  00.01   
214754                     80                      0.01  00.01   
214755                      2                      1.00    NaN   

                                         Descripción  ...  \
0                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
1                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
2                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
3                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
4                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
...                                              ...  ...   
214751  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214752  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214753  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214754  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214755                                           NaN  ...   

                                                 Capítulo  \
0       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
1       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
2       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
3       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
4       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
...                                                   ...   
214751  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214752  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214753  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214754  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214755                                                NaN   

       Código nuevo Establecimiento  \
0                           

Ahora se puede observar el nombre del procedimiento analizado y el nombre del hospital visto.


In [12]:
conteo_procedimientos_torax = ranking_con_nombre_procedimientos_y_establecimientos.query(
    "COD_HOSPITAL == @HOSPITAL_DEL_TORAX"
).sort_values(["ANIO_EGRESO", "conteo_procedimiento"], ascending=False)

mejores_procedimientos_torax = conteo_procedimientos_torax.query(
    "ranking_conteo_procedimiento == 1"
)

display(mejores_procedimientos_torax)

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo_procedimiento  \
54295        2021.0        112103         39.61                   626   
54251        2021.0        112103         39.64                   584   
52933        2021.0        112103         42.23                   391   
57293        2021.0        112103         37.22                   280   
13242        2021.0        112103         89.46                   278   
...             ...           ...           ...                   ...   
200802       2019.0        112103         34.89                     2   
201716       2019.0        112103         33.49                     2   
211026       2019.0        112103         07.81                     2   
198540       2019.0        112103         38.46                     1   
198589       2019.0        112103         38.42                     1   

        ranking_conteo_procedimiento  diferencias_procedimiento  \
54295                              1                         68   
54251                              1                        246   
52933                              1                        342   
57293                              1                         22   
13242                              1                         82   
...                              ...                        ...   
200802                             1                          1   
201716                             1                          1   
211026                             1                          1   
198540                             1                          0   
198589                             1                          0   

        totales_procedimiento  proporcion_procedimiento Código  \
54295                    3184                      0.20  39.61   
54251                    2020                      0.29  39.64   
52933                     607                      0.64  42.23   
57293                     801                      0.35  37.22   
13242                    1355                      0.21  89.46   
...                       ...                       ...    ...   
200802                      6                      0.33  34.89   
201716                     10                      0.20  33.49   
211026                     10                      0.20  07.81   
198540                      3                      0.33  38.46   
198589                      5                      0.20  38.42   

                                              Descripción  ...  \
54295   CIRCULACION EXTRACORPOREA PARA CIRUGIA CORAZON...  ...   
54251                  MARCAPASOS CORAZON INTRAOPERATORIO  ...   
52933                                  ESOFAGOSCOPIA OTRA  ...   
57293                  CATETERISMO CORAZON LADO IZQUIERDO  ...   
13242   COMPROBACION FORMA ONDA ARTEFACTO MARCAPASOS C...  ...   
...                                                   ...  ...   
200802                           OPERACION DIAFRAGMA.OTRA  ...   
201716        OPERACION PULMON REPARADORA Y PLASTICA OTRA  ...   
211026                         ESCISION TIMO PARCIAL OTRA  ...   
198540          RESECCION ARTERIA ABDOMEN CON SUSTITUCION  ...   
198589  RESECCION VASO CABEZA Y CUELLO CON SUSTITUCION...  ...   

                                                 Capítulo  \
54295   07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
54251   07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
52933   09. (42 54) OPERACIONES SOBRE EL APARATO DIGES...   
57293   07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
13242   16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
...                                                   ...   
200802  06. (30 34) OPERACIONES SOBRE EL APARATO RESPI...   
201716  06. (30 34) OPERACIONES SOBRE EL APARATO RESPI...   
211026  02. (06 07) OPERACIONES SOBRE EL SISTEMA ENDOC...   
198540  07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
198589  07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   

       

Los resultados muestran que el tórax en el año 2021 fue altamente relevante en los procedimientos
39.61, 39.64, 42.23, entre otros. Además, se puede observar la diferencia que tiene el Tórax
con los siguientes establecimientos de salud y el total de procedimientos realizados en el país.


Ahora, se quiere saber la cantidad de procedimientos, pero sabiendo si fue realizado de forma
ambulatoria, hospitalizada, cirugia mayor ambulatoria, etc...


In [14]:
var_categorica_a_contar = "procedimiento"
desglose_a_obtener = ["ANIO_EGRESO", "COD_HOSPITAL", "TIPO_ACTIVIDAD"]
grupo_ranking = ["ANIO_EGRESO", "TIPO_ACTIVIDAD", "procedimiento"]

conteo_y_ranking_procedimientos_por_tipoalta = (
    build_features.calcular_conteo_y_ranking_sobre_variable_categorica(
        procedimientos_formato_long,
        var_categorica_a_contar,
        desglose_a_obtener,
        grupo_ranking,
    )
)

In [28]:
with pd.ExcelWriter("../data/interim/procedimientos_grd.xlsx") as file:
    conteo_procedimientos_torax.to_excel(file, sheet_name="procedimientos_torax", index=False)
    conteo_y_ranking_procedimientos.to_excel(file, sheet_name="resumen_nacional", index=False)
    conteo_y_ranking_procedimientos_por_tipoalta.to_excel(
        file, sheet_name="resumen_nacional_tipoalta", index=False
    )

## 2. Obtención de perfiles de pacientes por procedimientos

En este apartado se quieren responder las siguientes preguntas:

1. ¿Cuál es la **mortalidad** de cada uno de los procedimientos?
2. ¿Cuál es el **peso GRD** promedio de cada uno de los procedimientos?
3. ¿Cuál es la **severidad** de cada uno de los procedimientos?
4. Dentro de los pacientes muertos, **¿Cuál es la severidad que tienen?**

Ahora, la base de GRD tiene hasta 30 procedimientos, por lo que en un mismo egreso se pueden
repetir procedimientos. Para reducir la complejidad del análisis, se utilizará solamente el primer
procedimiento para agrupar por severidad y mortalidad. Esto, ya que 1 procedimiento corresponderá
a 1 egreso.

Por lo tanto, la base de datos se debe mantener como está, pero solamente seleccionando las variables
recién mencionadas


In [16]:
vars_de_interes = [
    "COD_HOSPITAL",
    "ANIO_EGRESO",
    "TIPO_ACTIVIDAD",
    "CIP_ENCRIPTADO",
    "TIPOALTA",
    "DIAGNOSTICO1",
    "PROCEDIMIENTO1",
    "IR_29301_PESO",
    "IR_29301_SEVERIDAD",
]

In [17]:
df_para_mortalidad_y_severidad = df_procesada[vars_de_interes]

In [18]:
display(df_para_mortalidad_y_severidad)

COD_HOSPITAL  ANIO_EGRESO   TIPO_ACTIVIDAD  CIP_ENCRIPTADO  \
0              118100       2019.0  HOSPITALIZACIÓN       1314867.0   
1              118100       2019.0  HOSPITALIZACIÓN       1354418.0   
2              118100       2019.0  HOSPITALIZACIÓN        239861.0   
3              118100       2019.0  HOSPITALIZACIÓN        330326.0   
4              118100       2019.0  HOSPITALIZACIÓN       1369293.0   
...               ...          ...              ...             ...   
2750291        105100       2021.0  HOSPITALIZACIÓN      68882797.0   
2750292        120101       2021.0  HOSPITALIZACIÓN      97789022.0   
2750293        114101       2021.0  HOSPITALIZACIÓN      71164179.0   
2750294        107100       2021.0  HOSPITALIZACIÓN      78884679.0   
2750295        102100       2021.0  HOSPITALIZACIÓN      71716269.0   

                TIPOALTA DIAGNOSTICO1 PROCEDIMIENTO1  IR_29301_PESO  \
0              DOMICILIO        O82.0           74.1         0.5744   
1              DOMICILIO        O26.9          88.78         0.2951   
2              DOMICILIO        K92.0          87.03         0.6736   
3              DOMICILIO        A08.3          90.92         0.5475   
4              DOMICILIO        O60.0          75.34         0.3107   
...                  ...          ...            ...            ...   
2750291        DOMICILIO        N39.0          57.17         1.6547   
2750292        DOMICILIO        K40.9          53.00         0.8122   
2750293        DOMICILIO        Z51.8          00.18         1.0867   
2750294        DOMICILIO        K80.0          51.23         1.1472   
2750295  ALTA VOLUNTARIA          N10          87.71         0.5701   

         IR_29301_SEVERIDAD  
0                       2.0  
1                       1.0  
2                       2.0  
3                       2.0  
4                       1.0  
...                     ...  
2750291                 3.0  
2750292                 1.0  
2750293                 2.0  
2750294                 2.0  
2750295                 2.0  

[2750296 rows x 9 columns]

Ahora que se tiene la base de datos con menos variables, se quiere responder a la pregunta:

¿Cuál es la mortalidad de cada uno de los procedimientos?


In [19]:
desglose_procedimientos = df_para_mortalidad_y_severidad.groupby(
    [
        "COD_HOSPITAL",
        "ANIO_EGRESO",
        "TIPO_ACTIVIDAD",
        "TIPOALTA",
        "PROCEDIMIENTO1",
        "IR_29301_SEVERIDAD",
    ]
).agg(cantidad_procedimientos=("PROCEDIMIENTO1", "count"), peso_grd_medio=("IR_29301_PESO", "mean"))

display(desglose_procedimientos.head(10))

desglose_procedimientos = desglose_procedimientos.reset_index().sort_values(
    ["COD_HOSPITAL", "ANIO_EGRESO", "PROCEDIMIENTO1"]
)

cantidad_procedimientos  \
COD_HOSPITAL ANIO_EGRESO TIPO_ACTIVIDAD                  TIPOALTA  PROCEDIMIENTO1 IR_29301_SEVERIDAD                            
101100       2019.0      CIRUGÍA MAYOR AMBULATORIA (CMA) DOMICILIO 03.92          0.0                                      36   
                                                                   04.11          0.0                                       1   
                                                                   04.43          0.0                                      45   
                                                                   04.81          0.0                                       9   
                                                                   06.31          0.0                                       1   
                                                                   06.7           0.0                                       1   
                                                                   08.01          0.0                                       1   
                                                                   08.20          0.0                                       5   
                                                                   08.22          0.0                                       1   
                                                                   08.33          0.0                                       1   

                                                                                                      peso_grd_medio  
COD_HOSPITAL ANIO_EGRESO TIPO_ACTIVIDAD                  TIPOALTA  PROCEDIMIENTO1 IR_29301_SEVERIDAD                  
101100       2019.0      CIRUGÍA MAYOR AMBULATORIA (CMA) DOMICILIO 03.92          0.0                       0.148200  
                                                                   04.11          0.0                       0.297300  
                                                                   04.43          0.0                       0.304500  
                                                                   04.81          0.0                       0.147722  
                                                                   06.31          0.0                       0.987000  
                                                                   06.7           0.0                       0.552400  
                                                                   08.01          0.0                       0.188500  
                                                                   08.20          0.0                       0.297580  
                                                                   08.22          0.0                       0.273700  
                                                                   08.33          0.0                       0.491900

Los resultados anteriores indican lo siguiente:

El hospital 101100 en el año 2019 tuvo 36 procedimientos 1 03.92 que tuvieron una severidad de 0.
Estos procedimientos fueron realizados en una cirugía mayor ambulatorio (CMA), y la alta de
sus pacientes fueron en DOMICILIO. Dentro de los 36 egresos, estos tuvieron una complejidad
promedio de 0.148200.

Ahora, para obtener la mortalidad de cada procedimiento es necesario consultar los pacientes
que tienen un TIPOALTA igual a FALLECIDO.


In [20]:
display(desglose_procedimientos.query("TIPOALTA == 'FALLECIDO'"))

COD_HOSPITAL  ANIO_EGRESO   TIPO_ACTIVIDAD   TIPOALTA PROCEDIMIENTO1  \
1748          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          00.66   
1749          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          00.66   
1750          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          01.24   
1751          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          01.27   
1752          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          01.31   
...              ...          ...              ...        ...            ...   
343286        133150       2021.0  HOSPITALIZACIÓN  FALLECIDO          99.26   
343287        133150       2021.0  HOSPITALIZACIÓN  FALLECIDO          99.26   
343288        133150       2021.0  HOSPITALIZACIÓN  FALLECIDO          99.29   
343289        133150       2021.0  HOSPITALIZACIÓN  FALLECIDO          99.29   
343290        133150       2021.0  HOSPITALIZACIÓN  FALLECIDO          99.29   

        IR_29301_SEVERIDAD  cantidad_procedimientos  peso_grd_medio  
1748                   2.0                        2        1.818300  
1749                   3.0                        3        2.484000  
1750                   3.0                        1       11.704500  
1751                   2.0                        1        1.361400  
1752                   3.0                        1        4.345400  
...                    ...                      ...             ...  
343286                 2.0                        1        0.700000  
343287                 3.0                        2        1.622900  
343288                 1.0                        1        0.534600  
343289                 2.0                        8        1.203313  
343290                 3.0                       16        1.219869  

[22278 rows x 8 columns]

Los resultados anteriores indican que en el Hospital 101100, en el año 2019 fallecieron 2
egresos que se realizaron 00.66 en una hospitalización. Estos egresos tenían una severidad de 2,
con un peso GRD promedio de 1.8183.

A modo de ejemplo, se realizará la consulta en la base de datos y se mostrarán los pacientes
que cumplen los criterios.


In [21]:
df_procesada.query(
    "ANIO_EGRESO == 2019.0 and COD_HOSPITAL == 101100 and "
    "PROCEDIMIENTO1 == '00.66' and IR_29301_SEVERIDAD == 2.0 and TIPOALTA == 'FALLECIDO'"
)

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO  ETNIA  \
182374        101100       1370515.0  HOMBRE       1949-06-06  OTRO    
631299        101100        569058.0  HOMBRE       1937-12-16  OTRO    

       PROVINCIA COMUNA NACIONALIDAD                       PREVISION  \
182374     ARICA  ARICA        CHILE  FONASA INSTITUCIONAL - (MAI) D   
631299     ARICA  ARICA        CHILE  FONASA INSTITUCIONAL - (MAI) B   

       SERVICIO_SALUD  ... USOSPABELLON IR_29301_COD_GRD IR_29301_PESO  \
182374          ARICA  ...          4.0          51402.0        1.8183   
631299          ARICA  ...          4.0          51402.0        1.8183   

       IR_29301_SEVERIDAD IR_29301_MORTALIDAD HOSPPROCEDENCIA ESTANCIA  \
182374                2.0                 2.0             NaN     11.0   
631299                2.0                 2.0             NaN      3.0   

       ANIO_EGRESO MES_EGRESO    FECHA  
182374      2019.0       11.0  2019-11  
631299      2019.0       11.0  2019-11  

[2 rows x 133 columns]

Sin embargo, es necesario saber cuánto es el porcentaje de muertos del total.


In [22]:
total_proceds = desglose_procedimientos.groupby(["COD_HOSPITAL", "ANIO_EGRESO", "PROCEDIMIENTO1"])[
    "cantidad_procedimientos"
].transform("sum")

desglose_procedimientos["total_de_procedimientos"] = total_proceds

desglose_procedimientos["ratio_de_procedimientos"] = round(
    desglose_procedimientos["cantidad_procedimientos"]
    / desglose_procedimientos["total_de_procedimientos"],
    2,
)

In [23]:
display(desglose_procedimientos.head(10))

COD_HOSPITAL  ANIO_EGRESO   TIPO_ACTIVIDAD   TIPOALTA PROCEDIMIENTO1  \
414         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.17   
415         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.17   
416         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.66   
417         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.66   
418         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.66   
1748        101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          00.66   
1749        101100       2019.0  HOSPITALIZACIÓN  FALLECIDO          00.66   
419         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.70   
420         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.73   
421         101100       2019.0  HOSPITALIZACIÓN  DOMICILIO          00.73   

      IR_29301_SEVERIDAD  cantidad_procedimientos  peso_grd_medio  \
414                  1.0                        2          0.4149   
415                  2.0                        1          0.5390   
416                  1.0                       15          1.7184   
417                  2.0                       12          1.8183   
418                  3.0                        2          2.4840   
1748                 2.0                        2          1.8183   
1749                 3.0                        3          2.4840   
419                  2.0                        1          2.2875   
420                  1.0                        3          1.4477   
421                  2.0                        1          1.5876   

      total_de_procedimientos  ratio_de_procedimientos  
414                         3                     0.67  
415                         3                     0.33  
416                        34                     0.44  
417                        34                     0.35  
418                        34                     0.06  
1748                       34                     0.06  
1749                       34                     0.09  
419                         1                     1.00  
420                         4                     0.75  
421                         4                     0.25

Los resultados anteriores se pueden interpretar de la siguiente manera:

- El hospital 101100, en el año 2019 tuvo 5 procedimientos (y por ende 5 egresos) en donde
  la persona falleció realizandose un procedimiento 00.66. Estos egresos tenían una severidad 2
  y 3, y corresponden al 15% de todos los procedimientos realizados.


In [24]:
desglose_procedimientos = desglose_procedimientos.merge(
    cie_9, how="left", left_on="PROCEDIMIENTO1", right_on="Código"
).merge(
    esquema_hospitales,
    how="left",
    left_on="COD_HOSPITAL",
    right_on="Código nuevo Establecimiento",
)

In [25]:
desglose_procedimientos.to_csv(
    "../data/interim/mortalidad_y_severidad_procedimientos.csv",
    index=False,
    encoding="latin-1",
    sep=";",
    decimal=","
)